kits/agp1/test/agp1.js

```diff
@@ -0,0 +1,256 @@
+
+            const checkRole = async (appAddress, permission, managerAddress, appName='', roleName='', granteeAddress=managerAddress) => {
+                assert.equal(await acl.getPermissionManager(appAddress, permission), managerAddress, `${appName} ${roleName} Manager should match`)
+                assert.isTrue(await acl.hasPermission(granteeAddress, appAddress, permission), `Grantee should have ${appName} role ${roleName}`)
+            }
+
+            // app manager role
+            await checkRole(daoAddress, await dao.APP_MANAGER_ROLE(), metaTrackVoting.address, 'Kernel', 'APP_MANAGER')
+
+            // create permissions role
+            await checkRole(acl.address, await acl.CREATE_PERMISSIONS_ROLE(), metaTrackVoting.address, 'ACL', 'CREATE_PERMISSION')
+
+            // evm script registry
+            const regConstants = await getContract('EVMScriptRegistryConstants').new()
+            const reg = await getContract('EVMScriptRegistry').at(await acl.getEVMScriptRegistry())
+            assert.equal(await acl.getPermissionManager(reg.address, await reg.REGISTRY_ADD_EXECUTOR_ROLE()), NO_ADDRESS, 'EVMScriptRegistry ADD_EXECUTOR Manager should match')
+            assert.equal(await acl.getPermissionManager(reg.address, await reg.REGISTRY_MANAGER_ROLE()), NO_ADDRESS, 'EVMScriptRegistry REGISTRY_MANAGER Manager should match')
+
+            // voting
+            await checkRole(voting.address, await voting.CREATE_VOTES_ROLE(), owner, 'Voting', 'CREATE_VOTES', owner)
+            await checkRole(voting.address, await voting.MODIFY_QUORUM_ROLE(), metaTrackVoting.address, 'Voting', 'MODIFY_QUORUM')
+            await checkRole(voting.address, await voting.MODIFY_SUPPORT_ROLE(), metaTrackVoting.address, 'Voting', 'MODIFY_SUPPORT')
+
+            // voting meta track
+            await checkRole(metaTrackVoting.address, await metaTrackVoting.CREATE_VOTES_ROLE(), owner, 'MetaTrackVoting', 'CREATE_VOTES', owner)
+            await checkRole(metaTrackVoting.address, await metaTrackVoting.MODIFY_QUORUM_ROLE(), metaTrackVoting.address, 'MetaTrackVoting', 'MODIFY_QUORUM')
+            await checkRole(metaTrackVoting.address, await metaTrackVoting.MODIFY_SUPPORT_ROLE(), metaTrackVoting.address, 'MetaTrackVoting', 'MODIFY_SUPPORT')
+
+            // vault
+            await checkRole(vault.address, await vault.TRANSFER_ROLE(), metaTrackVoting.address, 'Vault', 'TRANSFER', finance.address)
+
+            // finance
+            await checkRole(finance.address, await finance.CREATE_PAYMENTS_ROLE(), metaTrackVoting.address, 'Finance', 'CREATE_PAYMENTS', voting.address)
+            await checkRole(finance.address, await finance.EXECUTE_PAYMENTS_ROLE(), metaTrackVoting.address, 'Finance', 'EXECUTE_PAYMENTS', voting.address)
+            await checkRole(finance.address, await finance.MANAGE_PAYMENTS_ROLE(), metaTrackVoting.address, 'Finance', 'MANAGE_PAYMENTS', voting.address)
+        })
+
+        context('creating vote', () => {
```

Comment:
Can we test this for both voting apps using a generative test?

Reply:
I would also add a test for how the metaTrackVoting app can execute meta actions, but it would revert if the voting app tries to perform those actions.